src/include/simeng/arch/aarch64/helpers/float.hh

```diff
@@ -157,6 +157,12 @@ RegisterValue scvtf_FixedPoint(
  * Returns single value of type D. */
 template <typename D, typename N>
 D fcvtzu_integer(srcValContainer& sourceValues) {
+  // Ensure types so that we know behaviour of inaccurate type conversions
+  static_assert((std::is_same<float, N>() || std::is_same<double, N>()) &&
```

Comment:
From looking around, fixed width floats are a >= C++23 feature (we are currently C++17), https://en.cppreference.com/w/cpp/types/floating-point . It seems to get a bit more complicated when you look into it with float and double e.g. https://www.quora.com/Can-we-consider-a-float-in-C-to-be-always-32-bit-wide-or-are-there-significant-exceptions as "The standard doesn't specify that a float be 32 bits wide or even that they by IEEE-754 ... the standard provides no guarantee" but " float and double rarely do [vary in size] (never do in my experience)".

This may be some of the cause to the problems we have been seeing with the `fcvtzu` tests. One suggestion in that thread is to do ` static_assert(sizeof(float) == 4 && CHAR_BIT == 8 && std::numeric_limits<float>::is_iec559);` so we know what we are dealing with. Some of the fixes in this PR do rely on knowing the underlying implementation of floats so I will looking into adding something along these lines